### PR TITLE
Fix a bug in MitreAttackV2

### PR DESCRIPTION
**Pull request by AradCarmi**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
relates: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-5596)

## Description
A fix for an issue when extracting the date from the description is failing.

### CODE REVIEW

**Comment by AradCarmi** [comment]
image
image

**Comment by AradCarmi** [comment]
After a short discussion with @MarkShleifer, we decided to change the `publications.timestamp` field format from `%Y, %b %d` to `%Y-%m-%dT%H:%M:%S` in order to support all server versions.

File: Packs/FeedMitreAttackv2/ReleaseNotes/1_1_10.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### MITRE ATT&CK
+- Updated the Docker image to: *demisto/taxii2:1.0.0.48199*.
+- Fixed an issue where the **fetch-indicators** command fails parsing publication description dates correctly.
```
Comment (dantavori) [inline Packs/FeedMitreAttackv2/ReleaseNotes/1_1_10.md:5]:
```suggestion
- Fixed an issue where the ***fetch-indicators*** command parses publication description dates incorrectly.
```

**Review by dantavori — APPROVED** [review]
nice

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2_test.py [hunk 2]
```diff
@@ -223,13 +223,29 @@ def test_filter_attack_pattern_object_by_attack_id(attack_id, attack_pattern_obj
 
 
 @pytest.mark.parametrize('description, expected_result', [
-    ('Gross, J. (2016, February 23). Operation Dust Storm. Retrieved December 22, 2021.', '2016, February 23'),
+    ('Test (23)', '1970-01-23T00:00:00'),
```
[the remark this answers is not in the document]
Comment (Ni-Knight) [reply Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2_test.py:226]: Should be 1970-01-01T00:00:00 so its clear it's not an actual date.

**Review by Ni-Knight — COMMENTED** [review]
Just one comment

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2_test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dantavori) [reply Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2_test.py:226]: If it's not a date, let's skip it and try the next match, if exists. Return 1970-01-01T00:00:00 only if nothing was found.

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py [hunk 3]
```diff
@@ -322,12 +324,21 @@ def map_fields_by_type(indicator_type: str, indicator_json: dict):
...
-        return ''
-    match = re.search(EXTRACT_TIMESTAMP_REGEX, description)
-    timestamp = match.group(1) if match else ''
-    return timestamp
+        return date_time_result
+    matches = re.findall(EXTRACT_TIMESTAMP_REGEX, description)
+    for match in matches:
+        try:
+            int(match)
+            continue
```
Comment (dantavori) [inline Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py:335]: please add a doc for this method, and explain the logic, including these lines.

**Review by dantavori — APPROVED** [review]

**Pull request merged**